### PR TITLE
Feature/11888 footer

### PR DESCRIPTION
**Pull request by erochOnGit**

### CODE REVIEW

File: src/web/theme/modules/Footer/EnhanceFooter.js [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
+      categoryId: CATEGORY_ID
+    })),
+    graphql(FooterQuery, {
+      options: props => ({
+        variables: {
+          id: props.categoryId
+        }
+      }),
+      props: ({ data }) =>
+        console.log("data", data) || {
```
Comment (milkae) [inline src/web/theme/modules/Footer/EnhanceFooter.js]: To be removed.

File: src/web/theme/modules/Footer/EnhanceFooter.js [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
+import withProps from "recompose/withProps";
+
+const CATEGORY_ID = 49;
 
 export default FooterQuery =>
-  graphql(FooterQuery, {
-    props: ({ data }) => ({
-      store: data.store
+  compose(
+    withProps(props => ({
```
Comment (milkae) [inline src/web/theme/modules/Footer/EnhanceFooter.js]: If you don't use it in the component, you should not pass it as a prop.

File: src/web/theme/modules/Footer/Footer.js [hunk 3]
```diff
@@ -4,13 +4,43 @@ import StoreDetails from "../StoreDetails";
...
 import EnhanceFooter from "./EnhanceFooter";
 import FooterQuery from "./FooterQuery.gql";
+import LoadingArea from "theme/ui/molecules/LoadingArea";
+import Logo from "theme/ui/atoms/Logo";
 import "./Footer.scss";
+import Mountain from "./Mountains.png";
 
-const Footer = ({ store }) => {
+const Footer = ({ loading, category }) => {
+  console.log(Mountain);
```
Comment (milkae) [inline src/web/theme/modules/Footer/Footer.js]: To be removed... please review your code before submitting it.

File: src/web/theme/modules/Footer/EnhanceFooter.js [hunk 4]
```diff
@@ -1,8 +1,24 @@
...
+    graphql(FooterQuery, {
+      options: props => ({
+        variables: {
+          id: props.categoryId
+        }
+      }),
+      props: ({ data }) =>
+        console.log("data", data) || {
+          loading: data.loading,
+          category: data.category
```
Comment (milkae) [inline src/web/theme/modules/Footer/EnhanceFooter.js]: I would have given a default value to the category, preventing errors in the component if it's null or undefined.

File: src/web/theme/modules/Footer/Footer.scss [hunk 5]
```diff
@@ -1,6 +1,65 @@
...
   color: $reverseFontColor;
   text-align: center;
   padding: 1em;
+  display: flex;
+  flex-direction: column;
+  margin-top: $moutainHeight;
+  font-family: $secondaryFontFamily;
+
+  & > div {
+    padding: 3em 3rem;
```
Comment (milkae) [inline src/web/theme/modules/Footer/Footer.scss]: Why the mixing ?

File: src/web/theme/modules/Footer/Footer.scss [hunk 6]
```diff
@@ -1,6 +1,65 @@
...
+    font-size: $small;
+  }
+
+  .mountain {
+    height: 164px;
+    width: 272px;
+    position: absolute;
+    margin-top: -$moutainHeight;
+    right: $moutainHeight;
+    z-index: -1;
```
Comment (milkae) [inline src/web/theme/modules/Footer/Footer.scss:61]: I'm not sure what you are trying to do, those CSS rules seem weird. Maybe we should look if there is a less hacky way to achieve your goal.

**Review by milkae — CHANGES_REQUESTED** [review]

File: src/web/theme/modules/Footer/Footer.js [hunk 7]
```diff
@@ -4,13 +4,43 @@ import StoreDetails from "../StoreDetails";
...
+        ) : (
+          category.children.map(category => (
+            <div className="category-list-title" key={category.id}>
+              {category.name}
+            </div>
+          ))
+        )}
+      </div>
+      <div className="footer__copyright">
+        Copyright 2018 - Blog & Shop - All right reserved - Developped by Blog &
```
Comment (hihuz) [inline src/web/theme/modules/Footer/Footer.js]: You should use `{new Date().getFullYear()}` instead of hard coding the year

Comment (hihuz) [reply src/web/theme/modules/Footer/Footer.js]: Also 'developed' with a single p :grinning:

File: src/web/theme/modules/Footer/Footer.js [hunk 8]
```diff
@@ -4,13 +4,43 @@ import StoreDetails from "../StoreDetails";
...
+          <span
+            dangerouslySetInnerHTML={{
+              __html: category.description
+            }}
+          />
+        )}
+      </div>
+      <div className="footer__categories">
+        {loading ? (
+          <LoadingArea>Loading subCategories</LoadingArea>
```
Comment (hihuz) [inline src/web/theme/modules/Footer/Footer.js]: I don't think the mention "subCategories" is particularly useful here, a plain "Loading" would be better

File: src/web/theme/modules/Footer/Footer.js [hunk 9]
```diff
@@ -4,13 +4,43 @@ import StoreDetails from "../StoreDetails";
...
+        ) : (
+          category.children.map(category => (
+            <div className="category-list-title" key={category.id}>
```
Comment (hihuz) [inline src/web/theme/modules/Footer/Footer.js]: I think it would be cleaner to follow BEM convention here, why not "footer__categories__title" ?

**Review by hihuz — CHANGES_REQUESTED** [review]
A couple of minor changes and this will be good to go :grin:

File: src/web/theme/modules/Footer/Footer.scss [hunk 10]
```diff
@@ -1,6 +1,65 @@
...
+  background: $brandSecondaryBackground;
   color: $reverseFontColor;
   text-align: center;
   padding: 1em;
+  display: flex;
+  flex-direction: column;
+  margin-top: $moutainHeight;
+  font-family: $secondaryFontFamily;
+
+  & > div {
```
Comment (real34) [inline src/web/theme/modules/Footer/Footer.scss:13]: Is the `div` really important here, or could it be any element? (hence to be replaced with `*`)

File: src/web/theme/modules/Footer/Footer.scss [hunk 11]
```diff
@@ -1,6 +1,65 @@
...
+  }
+
+  &__categories {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-around;
+    flex-wrap: wrap;
+
+    .category {
+      &-list-title {
```
Comment (real34) [inline src/web/theme/modules/Footer/Footer.scss]: There’s something wrong with BEM here.

Either use `.category__list-title`, or `.category-list__title` (and move `.category` styles def top level) or `.categories__list__title`

File: src/web/theme/modules/Footer/Footer.js [hunk 12]
```diff
@@ -4,13 +4,43 @@ import StoreDetails from "../StoreDetails";
...
+              {category.name}
```
Comment (real34) [inline src/web/theme/modules/Footer/Footer.js]: Shouldn’t it be a link? What’s the goal of displaying categories for the sake of their name?

**Review by real34 — APPROVED** [review]

File: src/web/theme/modules/Footer/Footer.scss (hunk 10 above)
Comment (erochOnGit) [reply src/web/theme/modules/Footer/Footer.scss:13]: yes it is

**Pull request merged**